packages/grz-watchdog/workflow/rules/common.smk

```diff
@@ -163,6 +163,44 @@ def get_validation_state(wildcards):
     return flag
 
 
+def anchor(target_pattern, source):
+    """
+    If the 'target_pattern' file exists, returns 'ancient(source)'.
+    Otherwise, returns 'source'.
+    This prevents Snakemake from checking timestamps if the job is already done.
+    """
+
+    def inner(wildcards):
+        target = target_pattern.format(**wildcards)
+        # Handle dynamic inputs (lambdas/functions) or static strings
+        src = source(wildcards) if callable(source) else source
+
+        if os.path.exists(target):
+            return ancient(src)
+        return src
+
+    return inner
+
+
+def payload(target_pattern, source):
+    """
+    If the 'target_pattern' file exists, returns empty list [] (input not needed).
+    Otherwise, returns 'source'.
+    This prevents Snakemake from erroring on missing temp files if the job is done.
```

Comment:
Just for my own understanding, why would it matter if any temp files are missing if the submission is `Finished`? Unless you mean resuming a submission where previous steps had their temp files deleted but they are no longer needed.